SnowplowTracker/SnowplowTracker/Tracker.cs

```diff
@@ -0,0 +1,342 @@
+using SnowplowTracker.Events;
+using SnowplowTracker.Storage;
+using SnowplowTracker.Enums;
+
+namespace SnowplowTracker {
+	public class Tracker {
+
+		private Emitter emitter;
+		private Subject subject;
+		private Session session;
+		private string trackerNamespace;
+		private string appId;
+		private DevicePlatforms platform;
+		private bool base64Encoded;
+
+		private bool dataCollection = true;
+		private bool synchronous = false;
+	
+		/// <summary>
+		/// Initializes a new instance of the <see cref="SnowplowTracker.Tracker"/> class.
+		/// </summary>
+		/// <param name="emitter">Emitter used for all sending and storing operations.</param>
+		/// <param name="trackerNamespace">Tracker namespace.</param>
+		/// <param name="appId">App identifier.</param>
+		/// <param name="platform">The DevicePlatform the tracker is running on.</param>
+		/// <param name="base64Encoded">If set to <c>true</c> all unstructured events and contextes will be base64 encoded.</param>
+		public Tracker(Emitter emitter, Subject subject, Session session, string trackerNamespace, string appId, DevicePlatforms platform = null, 
+		               bool base64Encoded = true) {
+			// Preconditions
+			Utils.CheckArgument(emitter != null, "Emitter cannot be null.");
+
+			// Tracker Setup
+			this.emitter = emitter;
+			this.trackerNamespace = trackerNamespace;
+			this.appId = appId;
+			this.platform = (platform != null) ? platform : DevicePlatforms.Mobile;
+			this.base64Encoded = base64Encoded;
+			this.session = session;
+
+			// Boot the emitter..
+			this.emitter.Start ();
+		}
+
+		// --- Tracker Control Methods
+
+		/// <summary>
+		/// Pauses the event tracking.
+		/// </summary>
+		public void PauseEventTracking() {
+			dataCollection = false;
+			if (session != null) {
+				session.StopSessionChecker();
+			}
+			emitter.Stop();
+		}
+
+		/// <summary>
+		/// Resumes the event tracking.
+		/// </summary>
+		public void ResumeEventTracking() {
+			dataCollection = true;
+			if (session != null) {
+				session.StartSessionChecker();
+			}
+			emitter.Start();
+		}
+
+		/// <summary>
+		/// Sets the tracker to synchronous operation.
+		/// </summary>
+		public void SetSynchronous() {
+			synchronous = true;
+			this.emitter.synchronous = true;
+		}
+
+		/// <summary>
+		/// Sets the tracker to asynchronous operation.
+		/// </summary>
+		public void SetAsynchronous() {
+			synchronous = false;
+			this.emitter.synchronous = false;
+		}
+
+		// --- Event Tracking
+
+		/// <summary>
+		/// Track the specified event.
+		/// - Checks whether dataCollection is on
+		/// - Checks whether to use synchronous sending or not
+		/// </summary>
+		/// <param name="newEvent">New event to track.</param>
+		public void Track(IEvent newEvent) {
+			if (!dataCollection) {
+				return;
+			}
+			if (synchronous) {
+				ProcessEvent (newEvent);
+			} else {
+				ThreadPool.QueueUserWorkItem (new WaitCallback ( delegate(object t) {
+					ProcessEvent (newEvent);
```

Comment:
My reasoning for this was to move all of the processing for building the TrackerPayload into the background.  Removing any of the wait time on the UI Thread which is quite a useful thing in regards to making the library as performant as possible on mobile.

It also means that I made my concurrent queue for a reason!